symfony/plugins/orangehrmAdminPlugin/client/jobCategory/components/dialogs/DeleteConfirmationDialog.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <oxd-dialog v-model:show="show" :style="{ maxWidth: '450px' }">
+    <div class="orangehrm-modal-header">
+      <oxd-text type="card-title">Are you sure?</oxd-text>
+    </div>
+    <div class="orangehrm-text-center-align">
+      <oxd-text type="subtitle-2">
+        The selected item will be permanently deleted. Are you sure you want to
+        continue?
+      </oxd-text>
+    </div>
+    <div class="orangehrm-modal-footer">
+      <oxd-button
+        label="No, Cancel"
+        displayType="text"
+        class="orangehrm-button-margin"
+        @click="onCancel"
+      />
+      <oxd-button
+        label="Yes, Delete"
+        iconName="trash"
+        displayType="label-danger"
+        class="orangehrm-button-margin"
+        @click="onDelete"
+      />
+    </div>
+  </oxd-dialog>
+</template>
+
+<script>
+import Dialog from "@orangehrm/oxd/core/components/Dialog/Dialog";
+
+export default {
+  components: {
+    "oxd-dialog": Dialog,
+  },
+  data() {
+    return {
+      show: false,
+      reject: null,
+      resolve: null,
+    };
+  },
+  methods: {
+    showDialog() {
+      return new Promise((resolve, reject) => {
+        this.resolve = resolve;
+        this.reject = reject;
+        this.show = true;
+      });
+    },
+    onDelete() {
+      this.show = false;
+      this.resolve && this.resolve("ok");
```

Comment:
is it required to resolve with parameter?

Reply:
might be helpful in case where we have more than two options?